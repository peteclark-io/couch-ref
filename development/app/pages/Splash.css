@import "../css/branding.css";

.splash {
   position: absolute;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: var(--brand-primary)
}

.brand {
   font-family: var(--font-primary);
   color: white;
   font-size: 6em;
   text-align: center;
   margin-top: 30vh;
}

.loading {
   font-family: var(--font-secondary-bold);
   color: white;
   text-align: center;
   margin-top: 20px;
}
