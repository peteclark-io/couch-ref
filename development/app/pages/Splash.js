'use strict';

import React from 'react';
import { connect } from 'react-redux';
import {ThreeBounce} from 'better-react-spinkit';

import styles from './Splash.css';

const Splash = React.createClass({

   propTypes: {
      ready: React.PropTypes.bool
   },

   contextTypes: {
      router: React.PropTypes.object
   },

   componentWillMount: function(){
      this.checkIfReady();
   },

   componentWillUpdate: function(){
      this.checkIfReady();
   },

   checkIfReady: function(){
      if (this.props.ready){
         this.context.router.push('/');
      }
   },

   render: function() {
     return (
      <div className={styles.splash}>
         <h1 className={styles.brand}>Couch Ref</h1>
         <div className={styles.loader}>
            <ThreeBounce color="white" />
         </div>
      </div>
    );
  }
});

const isReady = (state = {ready: false}) => {
   return state.ready;
};

const mapStateToProps = (state) => {
   return {
     ready: isReady(state)
   };
};

const LiveSplash = connect(
  mapStateToProps
)(Splash);

export default LiveSplash;
