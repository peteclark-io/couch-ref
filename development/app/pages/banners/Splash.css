@import "../../css/branding.css";

.splash {
   position: absolute;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: var(--brand-primary)
}

.brand {
   font-family: var(--font-primary);
   color: var(--brand-secondary);
   font-size: 6em;
   text-align: center;
   margin-top: 30vh;
}

.loader {
   color: var(--brand-secondary);
   text-align: center;
   margin-top: 20px;
}

.version {
   font-family: var(--font-primary);
   color: var(--brand-secondary);
   text-align: center;
   margin-top: 6vh;
}
