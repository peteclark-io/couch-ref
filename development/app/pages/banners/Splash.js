'use strict';

import React from 'react';
import { connect } from 'react-redux';
import {ThreeBounce} from 'better-react-spinkit';

import styles from './Splash.css';

const Splash = React.createClass({

   propTypes: {
      ready: React.PropTypes.bool,
      user: React.PropTypes.object
   },

   contextTypes: {
      router: React.PropTypes.object
   },

   componentWillMount: function(){
      this.checkIfReady();
   },

   componentWillUpdate: function(){
      this.checkIfReady();
   },

   checkIfReady: function(){
      if (this.props.ready){
         this.context.router.push('/');
      }
   },

   render: function() {
     return (
      <div className={styles.splash}>
         <h1 className={styles.brand}>Couch Ref</h1>
         <div className={styles.loader}>
            <ThreeBounce color="white" />
         </div>
      </div>
    );
  }
});

const isReady = (state = {ready: false}) => {
   return state.ready;
};

const getUser = (state = {user: {}}) => {
   return state.user;
};

const mapStateToProps = (state) => {
   return {
     ready: isReady(state),
     user: getUser(state)
   };
};

const LiveSplash = connect(
  mapStateToProps
)(Splash);

export default LiveSplash;
