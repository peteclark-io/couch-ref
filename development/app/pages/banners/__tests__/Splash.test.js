'use strict';

import React from 'react';
import {component} from '../Splash.js';
import {mount, shallow} from 'enzyme';

import renderer from 'react-test-renderer';

test('Splash', () => {
   const Splash = component();
   const rendered = shallow(
      <Splash ready={false} />
   );

   let tree = rendered.not('.loading').debug();
   expect(tree).toMatchSnapshot();
});

it('Redirects if ready', () => {
   const Splash = component();
   var routed = false;
   const router = {
      push: (uri) => {
         expect(uri).toEqual('/');
         routed = true;
      }
   };

   const rendered = shallow(
      <Splash ready={true} />,
      {context: {router: router}}
   );

   expect(routed).toBeTruthy();
});

it('Doesn\'t redirect if not ready', () => {
   const Splash = component();
   const router = {
      push: (uri) => {
         expect(false).toBeTruthy(); // aka, fail pls.
      }
   };

   const rendered = shallow(
      <Splash ready={false} />,
      {context: {router: router}}
   );
});
