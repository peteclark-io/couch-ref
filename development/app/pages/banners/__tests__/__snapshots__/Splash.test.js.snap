exports[`test Splash 1`] = `
"<div className=\"splash\">
<h1 className=\"brand\">
Couch Ref
</h1>
<div className=\"loader\">
<ThreeBounce color=\"white\" duration=\"1.5s\" size={10} scaleEnd={1} scaleStart={0} gutter={2} timingFunction=\"ease-in-out\" />
</div>
</div>"
`;
